bot/exts/fun/hangman.py

```diff
@@ -0,0 +1,181 @@
+        - singleplayer: writing 's' means you want to play by yourself, and only you can suggest letters,
+            - writing 'm' means you want multiple players to join in and guess the word.
+        """
+        # Changing singleplayer to a boolean
+        singleplayer = True if singleplayer.lower() == 's' else False
+
+        # Filtering the list of all words depending on the configuration
+        filtered_words = [
+            word for word in ALL_WORDS
+            if min_length < len(word) < max_length
+            and min_unique_letters < len(set(word)) < max_unique_letters
+        ]
+
+        if not filtered_words:
+            filter_not_found_embed = Embed(
+                title=choice(NEGATIVE_REPLIES),
+                description="No words could be found that fit all filters specified.",
+                color=Colours.soft_red,
+            )
+            await ctx.send(embed=filter_not_found_embed)
+            return
+
+        word = choice(filtered_words)
+
+        user_guess = "_" * len(word)
+        tries = 6
+        guessed_letters = set()
+
+        # Checking if the game is singleplayer
+        def check(msg: Message) -> bool:
+            if singleplayer:
+                return msg.author == ctx.author
+            else:
+                # Multiplayer mode
+                return msg.author != self.bot
+
+        hangman_embed = Embed(
+            title="Hangman",
+            color=Colours.python_blue,
+        )
+        hangman_embed.set_image(url=IMAGES[tries])
+        hangman_embed.add_field(
+            name=f"You've guessed `{user_guess}` so far.",
+            value="Guess the word by sending a message with the letter!",
+            inline=False,
+        )
+        hangman_embed.set_footer(text=f"Tries: {tries}")
+        original_message = await ctx.send(embed=hangman_embed)
+
+        while user_guess != word:
+            try:
+                message = await self.bot.wait_for(
+                    event="message",
+                    timeout=60.0,
+                    check=check
+                )
+            except TimeoutError:
+                timeout_embed = Embed(
+                    title=choice(NEGATIVE_REPLIES),
+                    description="Looks like the bot timed out! You must send a letter within 60 seconds.",
+                    color=Colours.soft_red,
+                )
+                await original_message.edit(embed=timeout_embed)
+                return
+
+            message.content = message.content.lower()
+            if len(message.content) > 1:
+                letter_embed = Embed(
+                    title=choice(NEGATIVE_REPLIES),
+                    description="You can only send a letter for the hangman game, try again!",
+                    color=Colours.dark_green,
+                )
+                to_delete = await ctx.send(embed=letter_embed)
+                await to_delete.delete(delay=4)
```

Comment:
TextChannel.send has the `delete_after` option which does the same

Reply:
I'll change to that - but is there any difference between the two?

Reply:
Not at all, `delete_after` spawns an task that sleeps, then deletes the message. This accomplishes the same thing

Reply:
So should I change it?

Reply:
IMO using delete after is cleaner but, feel free to leave it as it

Reply:
Yea, I prefer using the `delete_after` kwarg for simplicity too, but it doesn't matter either way really.